projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts

```diff
@@ -0,0 +1,168 @@
+
+@Component({
+  selector: 'xuilib-write-address-field',
+  styleUrls: ['./write-address.component.scss'],
+  templateUrl: './write-address.component.html'
+})
+export class WriteAddressFieldComponent implements OnInit, OnChanges {
+
+  @Input()
+  public formGroup: FormGroup;
+  @Input()
+  public internationalMode = false;
+  @Input()
+  public isInternational: boolean;
+  @Input()
+  public submissionAttempted = false;
+  @Input()
+  public startedInternational: boolean;
+
+  @Output() public postcodeOptionSelected = new EventEmitter<void>();
+  @Output() public internationalModeStart = new EventEmitter<void>();
+  @Output() public ukAddressOptionSelected = new EventEmitter<boolean>();
+
+  public addressesService: AddressService;
+
+  public addressField: AddressModel = {
+    addressLine1: '',
+    addressLine2: '',
+    addressLine3: '',
+    postCode: '',
+    postTown: '',
+    country: '',
+    county: ''
+  };
+
+  public optionErrorsPresent = false;
+
+  public addressChosen = false;
+
+  public addressFormGroup: FormGroup;
+
+  public addressOptions: AddressOption[];
+
+  public missingPostcode = false;
+
+  public optionErrorMessage = AddressMessageEnum.NO_OPTION_SELECTED;
+
+  constructor(addressesService: AddressService) {
+    this.addressesService = addressesService;
+  }
+
+  public ngOnInit(): void {
+    if (!this.formGroup.get('address')) {
+      this.formGroup.addControl('address', new FormControl({}));
+    }
+    // set the form group relevant to only the external parent component
+    this.addressFormGroup = new FormGroup({
+      // relevant to international mode
+      ukAddress: new FormControl(this.isInternational !== undefined ? (this.isInternational ? 'no' : 'yes') : null),
+      // relevant to postocode lookup
+      postcode: new FormControl(''),
+      addressList: new FormControl('')
+    });
+  }
+
+  public findAddress() {
+    if (!this.addressFormGroup.get('postcode').value) {
+      this.missingPostcode = true;
+    } else {
+      this.missingPostcode = false;
+      this.addressField = null;
+      const postcode = this.addressFormGroup.get('postcode').value;
+      this.addressOptions = [];
+      this.addressesService.getAddressesForPostcode(postcode.replace(' ', '').toUpperCase()).subscribe(
+        result => {
+          result.forEach(
+            address => {
+              this.addressOptions.push(new AddressOption(address, null));
+            }
+          );
+          this.addressOptions.unshift(
+            new AddressOption(undefined, this.defaultLabel(this.addressOptions.length))
+          );
+        }, (error) => {
+          console.log(`An error occurred retrieving addresses for postcode ${postcode}. ${error}`);
+          this.addressOptions.unshift(
+            new AddressOption(undefined, this.defaultLabel(this.addressOptions.length))
+          );
+        });
+      this.addressFormGroup.get('addressList').setValue(undefined);
+      // this.refocusElement();
```

Comment:
@connorpgpmcelroy - can we get rid of this commented line?

Reply:
Yes